components/setting_item_min/setting_item_min.tsx

```diff
@@ -1,79 +1,73 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
 import {isMobile} from 'utils/utils.jsx';
 import EditIcon from 'components/widgets/icons/fa_edit_icon';
 
-export default class SettingItemMin extends React.PureComponent {
-    static defaultProps = {
-        section: '',
-    };
-
-    static propTypes = {
-
-        /**
-         * Settings title
-         */
-        title: PropTypes.node,
-
-        /**
-         * Option to disable opening the setting
-         */
-        disableOpen: PropTypes.bool,
-
-        /**
-         * Settings or tab section
-         */
-        section: PropTypes.string,
-
-        /**
-         * Function to update section
-         */
-        updateSection: PropTypes.func,
-
-        /**
-         * Settings description
-         */
-        describe: PropTypes.node,
-
-        /**
-         * Shows the previous active section for focusing
-         */
-        previousActiveSection: PropTypes.string,
-
-        /**
-         * Actions
-         */
-        actions: PropTypes.shape({
-
-            /**
-             * Update the active section for focusing
-             */
-            updateActiveSection: PropTypes.func.isRequired,
-        }).isRequired,
-    };
+interface SettingItemMinProps {
```

Comment:
I suggests to name it as `Props` only unless there's more than once component in a file that needs to distinguish props. 
```suggestion
interface Props {
```